drivers/ksz8851snl/ksz8851_netdev.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * Copyright (C) 2015 Ell-i open source co-operative
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_ksz8851snl
+ * @{
+ *
+ * @file
+ * @brief       Internal functions for the ksz8851snl driver
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+
+#include "mutex.h"
+#include "ksz8851snl_regs.h"
+#include "ksz8851snl.h"
+
+#include "xtimer.h"
+
+#include "net/netdev2.h"
+#include "net/eui64.h"
+#include "net/ethernet.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+#define SECT_NAME "[KSZ8851 Driver]:"
+
+static void lock(ksz8851snl_t *dev) {
+	mutex_lock(&dev->mutex);
+}
+
+static void unlock(ksz8851snl_t *dev) {
+	mutex_unlock(&dev->mutex);
+}
+
+static void ksz_spi_transfer(ksz8851snl_t *dev, char *out, char *in, int len) {
+	spi_acquire(dev->spi);
+	gpio_clear(dev->cs);
+	spi_transfer_bytes(dev->spi, out, in, len);
+	gpio_set(dev->cs);
+	spi_release(dev->spi);
+}
+
+static void reg_set(ksz8851snl_t *dev, uint16_t reg, uint16_t value) {
+	uint16_t cmd = (reg << 2) & 0x3F0;//might add addr mask
```

Comment:
it would be nice if we would take the endianess of the host system into account here, I suggest either:
- use the byteorder types (core/include/byteorder.h)
- use something like
```
#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
...
#else
...
#endif
```
- or simply: add a todo comment like `/** @todo take care of host bytorder */